### PR TITLE
Add Metrics middleware

### PR DESCRIPTION
**Pull request by bryce-anderson**
Metrics can be done backend independently if we simple use it as a
middleware.

This implementation is modeled after the reference
metrics-jetty package.

### CODE REVIEW

File: examples/blaze/examples-blaze.sbt [hunk 1]
```diff
@@ -6,6 +6,10 @@ publishArtifact := false
 
 fork := true
 
+libraryDependencies ++= Seq(
+  "io.dropwizard.metrics" % "metrics-json" % "3.1.0"
```
Comment (rossabaker) [inline examples/blaze/examples-blaze.sbt]: Let's use the definition in Http4sBuild.  That centralizes our dependency declarations and keeps them in sync.

**Comment by rossabaker** [comment]
Aside from comment, :+1:

**Comment by bryce-anderson** [comment]
@rossabaker, I'd just merge this (once travis gives the OK), but I'm only 98% sure thats what you wanted.

**Comment by rossabaker** [comment]
:+1:

**Comment by dchenbecker** [comment]
Nice!

**Pull request merged**